rotkehlchen/chain/ethereum/airdrops.py

```diff
@@ -334,37 +337,49 @@ def _process_json(response: Response, filename: Path) -> None:
 def calculate_claimed_airdrops(
         airdrop_data: Sequence[AirdropClaimEventQueryParams],
         database: DBHandler,
-) -> Sequence[tuple[ChecksumEvmAddress, str, FVal]]:
+) -> Sequence[tuple[ChecksumEvmAddress, str]]:
     """Calculates which of the given airdrops have been claimed.
     It does so by checking if there is a claim history event in the database
     that matches the airdrop data (address, asset and amount). It returns a list
     of tuples with the airdrop data for which a claim event was found."""
     if len(airdrop_data) == 0:
         return []
 
-    query_parts, bindings = [], []
-    for airdrop_info in airdrop_data:
-        amount = airdrop_info.tolerance.value
-        half_range = airdrop_info.tolerance.tolerance / 2
-        query_parts.append('(events.type=? AND events.subtype=? AND events.location_label=? AND events.asset=? AND CAST(events.amount AS REAL) BETWEEN ? AND ?)')  # noqa: E501
-        bindings.extend(
-            [
-                airdrop_info.event_type.serialize(),
-                airdrop_info.event_subtype.serialize(),
-                airdrop_info.location_label,
-                airdrop_info.asset.serialize(),
-                str(amount - half_range),
-                str(amount + half_range),
-            ],
-        )
-
-    query_part = ' OR '.join(query_parts)
+    claimed_events = []
     with database.conn.read_ctx() as cursor:
-        return cursor.execute(
-            'SELECT events.location_label, events.asset, events.amount '
-            f'FROM history_events AS events WHERE {query_part}',
-            tuple(bindings),
-        ).fetchall()
+        for airdrop_info in airdrop_data:
+            amount = airdrop_info.tolerance.value
+            half_range = airdrop_info.tolerance.tolerance / 2
+
+            for address, db_extra_data, in cursor.execute(
+                'SELECT events.location_label,evm_events.extra_data FROM history_events AS events '
+                'LEFT JOIN evm_events_info AS evm_events ON events.identifier=evm_events.'
+                'identifier WHERE (events.type=? AND events.subtype=? AND events.location_label=? '
+                'AND events.asset=? AND CAST(events.amount AS REAL) BETWEEN ? AND ?);',
+                (
+                    airdrop_info.event_type.serialize(),
+                    airdrop_info.event_subtype.serialize(),
+                    airdrop_info.location_label,
+                    airdrop_info.asset.serialize(),
+                    str(amount - half_range),
+                    str(amount + half_range),
+                ),
+            ):
+                if db_extra_data is None:
+                    continue
```

Comment:
shouldn't we log an error message here? maybe even show a database.message_aggregator.add_warning error?